### PR TITLE
Improve dataset loading

### PR DESCRIPTION
**Pull request by pzhanggit**
1. fixed a bug when loading datasets separately for training/test/validate
2. remove the need that dataset has to be one of the prescribed in dataset_descriptors.py
3. fix a bug related to  https://github.com/ORNL/HydraGNN/pull/22

### CODE REVIEW

**Review by allaffa — APPROVED** [review]

File: hydragnn/preprocess/load_data.py [hunk 1]
```diff
@@ -23,65 +23,22 @@
 
 from hydragnn.preprocess.serialized_dataset_loader import SerializedDataLoader
 from hydragnn.preprocess.raw_dataset_loader import RawDataLoader
-from hydragnn.preprocess.dataset_descriptors import Dataset
```
Comment (streeve) [inline hydragnn/preprocess/load_data.py:26]: Let's remove if no longer using

**Review by streeve — COMMENTED** [review]
@pzhanggit you're not allowed to title "fix bug" anymore 👎

File: hydragnn/preprocess/load_data.py (hunk 1 above)
Comment (pzhanggit) [reply hydragnn/preprocess/load_data.py:26]: Removed in 5a35268f9739410d224d78e4d542b966b4870161

File: hydragnn/run_training.py [hunk 2]
```diff
@@ -77,7 +77,14 @@ def _(config: dict):
         and config["NeuralNetwork"]["Variables_of_interest"]["denormalize_output"]
         == "True"
     ):
-        dataset_path = f"{os.environ['SERIALIZED_DATA_PATH']}/serialized_dataset/{config['Dataset']['name']}.pkl"
+        for dataset_name, _ in config["Dataset"]["path"]["raw"].items():
```
[the remark this answers is not in the document]
Comment (pzhanggit) [reply hydragnn/run_training.py]: Done in 83fa1d1e03601f18c5b036a410d8cef71a83cd6f

File: hydragnn/run_prediction.py [hunk 3]
```diff
@@ -130,7 +131,9 @@ def _(config: dict):
     ##output predictions with unit/not normalized
     if config["NeuralNetwork"]["Variables_of_interest"]["denormalize_output"] == "True":
         true_values, predicted_values = output_denormalize(
-            config["Variables_of_interest"]["y_minmax"], true_values, predicted_values
+            config["NeuralNetwork"]["Variables_of_interest"]["y_minmax"],
```
Comment (streeve) [inline hydragnn/run_prediction.py:134]: Did this not fail because we don't test it?

File: hydragnn/run_training.py [hunk 4]
```diff
@@ -77,7 +77,11 @@ def _(config: dict):
         and config["NeuralNetwork"]["Variables_of_interest"]["denormalize_output"]
         == "True"
     ):
-        dataset_path = f"{os.environ['SERIALIZED_DATA_PATH']}/serialized_dataset/{config['Dataset']['name']}.pkl"
+        if "total" in config["Dataset"]["path"]["raw"].items():
+            dataset_path = f"{os.environ['SERIALIZED_DATA_PATH']}/serialized_dataset/{config['Dataset']['name']}.pkl"
+        else:
+            ###used for min/max values loading below
```
Comment (streeve) [inline hydragnn/run_training.py:83]: add "and so we only need to read one file"

File: hydragnn/run_prediction.py (hunk 3 above)
Comment (pzhanggit) [reply hydragnn/run_prediction.py:134]: It was added recently and is not tested in current unit test. For other runs, I do not know.

File: hydragnn/run_training.py (hunk 4 above)
Comment (pzhanggit) [reply hydragnn/run_training.py:83]: I think this is good enough.

**Pull request merged**